ui/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -134,14 +134,22 @@ const mapStateToProps = (state, ownProps) => {
   const { txParams = {}, id: transactionId, type } = txData;
   const txId = transactionId || paramsTransactionId;
   const transaction = getUnapprovedTransaction(state, txId);
+  const finalTxParams = (transaction && transaction.txParams) || txParams;
+
+  if (!Object.keys(finalTxParams).length) {
+    // finalTxParams is empty. We don't have enough information to render this component.
+    // Fixes https://github.com/MetaMask/metamask-extension/issues/21026
```

Comment:
While additional audit information is great, would this pattern add a lot of additional comments to the codebase over time?

Each commit already includes the PR number in the title meaning we can always use git blame to come back to the PR and verify any context and related issues.